stl/inc/syncstream

```diff
@@ -218,11 +218,14 @@ protected:
         }
 
         const _Size_type _New_capacity  = _Calculate_growth(_Buf_size, _Buf_size + 1, _Max_allocation);
-        const _Elem* const _Old_ptr     = streambuf_type::pbase();
+        _Elem* const _Old_ptr           = streambuf_type::pbase();
         const _Size_type _Old_data_size = _Get_data_size();
 
         _Elem* const _New_ptr = _Unfancy(_Al.allocate(_New_capacity));
         _Traits::copy(_New_ptr, _Old_ptr, _Old_data_size);
+        if (0 < _Buf_size) {

```

Comment:
That is a Yoda condition and I believe `_Buf_size` should be unsigned, so checking `!=` might be better

Reply:
We use Yoda conditions everywhere in iostreams, in other include files too:
https://github.com/microsoft/STL/blob/17fde2cbab6e8724d81c9555237c9a623d7fb954/stl/inc/streambuf#L337

But yes, `_Buf_size` is unsigned.
https://timsong-cpp.github.io/cppwp/n3337/allocator.requirements#tab:utilities.allocator.requirements
```
X::size_type 	unsigned integer type 	a type that can represent the size of the largest object in the allocation model. 
```